cloudify_aws/ec2/elasticip.py

```diff
@@ -0,0 +1,300 @@
+                constants.EXTERNAL_RESOURCE_ID]
+        vpc_id = ctx.source.instance.runtime_properties.get('vpc_id')
+        if vpc_id:
+            ctx.target.instance.runtime_properties['vpc_id'] = vpc_id
+
+        return True
+
+    def disassociate(self, args=None, **_):
+        """ Disassocates an Elastic IP created by Cloudify from an EC2 Instance
+        that was also created by Cloudify.
+        """
+
+        elasticip = \
+            utils.get_external_resource_id_or_raise(
+                    'disassociate elasticip', ctx.target.instance)
+
+        elasticip_object = self._get_address_object_by_id(elasticip)
+
+        if not elasticip_object:
+            raise NonRecoverableError(
+                    'no matching elastic ip in account: {0}'.format(elasticip))
+
+        disassociate_args = dict(
+                public_ip=elasticip_object.public_ip,
+                association_id=elasticip_object.association_id
+        )
+
+        try:
+            self.execute(self.client.disassociate_address,
+                         disassociate_args, raise_on_falsy=True)
+        except (exception.EC2ResponseError,
+                exception.BotoServerError) as e:
+            raise NonRecoverableError('{0}'.format(str(e)))
+
+        return True
+
+    def post_disassociate(self):
+        super(ElasticIPInstanceConnection, self).post_disassociate()
+        utils.unassign_runtime_property_from_resource(
+                'public_ip_address', ctx.source.instance)
+        utils.unassign_runtime_property_from_resource(
+                'instance_id', ctx.target.instance)
+        if ctx.source.instance.runtime_properties.get('vpc_id'):
+            utils.unassign_runtime_property_from_resource(
+                'vpc_id', ctx.target.instance)
+
+        return True
+
+    def _get_address_object_by_id(self, address_id):
+        """Returns the elastip object for a given address elastip.
+
+        :param address_id: The ID of a elastip.
+        :returns The boto elastip object.
+        """
+
+        address = self._get_all_addresses(address=address_id)
+
+        return address[0] if address else address
+
+    def _get_all_addresses(self, address=None):
+        """Returns a list of elastip objects for a given address elastip.
+
+        :param address: The ID of a elastip.
+        :returns A list of elasticip objects.
+        :raises NonRecoverableError: If Boto errors.
+        """
+
+        try:
+            addresses = self.client.get_all_addresses(address)
+        except exception.EC2ResponseError as e:
+            if constants.ELASTICIP['NOT_FOUND_ERROR'] in e:
+                addresses = self.client.get_all_addresses()
+                utils.log_available_resources(addresses)
+            return None
+        except exception.BotoServerError as e:
+            raise NonRecoverableError('{0}'.format(str(e)))
+
+        return addresses
+
+    def get_source_resource(self):
```

Comment:
why can't you use https://github.com/cloudify-cosmo/cloudify-aws-plugin/blob/master/cloudify_aws/base.py#L241